Source/WebKit/Platform/haiku/SharedMemoryHaiku.cpp

```diff
@@ -58,19 +60,44 @@ bool SharedMemory::Handle::decode(IPC::Decoder& decoder, Handle& handle)
 {
 }
 
-static int createSharedMemory()
+static uint32 protectionMode(SharedMemory::Protection protection)
 {
+	switch(protection)
+	{
+		case SharedMemory::Protection::ReadOnly:
+		return B_READ_AREA;
+		case SharedMemory::Protection::ReadWrite:
+		return B_READ_AREA | B_WRITE_AREA;
+	}
 }
 
 RefPtr<SharedMemory> SharedMemory::allocate(size_t size)
 {
+	void* baseAddress;
+	fprintf(stderr,"%s %ld\n",__PRETTY_FUNCTION__,size);
+	//size = ROUND_UP_TO_PAGE(size); if its not a multiple of a page then do this
+	area_id sharedArea = create_area("WebKit-Shared-Memory",&baseAddress,B_ANY_ADDRESS,
+		size,B_NO_LOCK,B_READ_AREA | B_WRITE_AREA);
+	
+	if(sharedArea<0)
+	return nullptr;
+	
+	RefPtr<SharedMemory> memory = adoptRef(new SharedMemory);
+	memory->m_size = size;
+	memory->m_data = baseAddress;
+	memory->m_areaid = sharedArea;
 }
 
 RefPtr<SharedMemory> SharedMemory::map(const Handle& handle, Protection protection)
 {
 
```

Comment:
we will need this working soon. Have you read about areas and how they are used to share memory between two applications?

Reply:
I guess they are used for bitmaps for now they are not important as we have to fix messages first

Reply:
I'm not sure, then at least log something here so we know we have reached the point where webkit is trying to use this